### ISSUE TITLE
Dice and Iou

### ISSUE THREAD

**Issue by closertt** [atlan-antillia/Image-Segmentation-Gastrointestinal-Polyp#3]
Hello, can you provide the function to calculate Dice and Iou in the code, these evaluation indicators are very important in semantic segmentation.

**Comment by atlan-antillia** [comment]
Hi.
Thank you so much for emailing me.
As you know or don't know,  some typical loss-functions including Dice loss
are already in losses.py in our repositories.

For example, please refer to the following source code.
https://github.com/atlan-antillia/Tiled-Image-Segmentation-Multiple-Myeloma/blob/main/losses.py

Good luck, and thank you.
Have a nice weekend!


2023年6月24日(土) 21:18 closertt ***@***.***>:

**Comment by atlan-antillia** [comment]
I have just added the following explanation.
If you would like dice_loss function,
1 Modify train_eval_infer.config as below.
[model]
;loss           = "bce_iou_loss"
**loss           = "dice_loss"**

2 Modify TensorflowUNet.py to add the following line.
**from losses import dice_loss**

**Comment by closertt** [comment]
Hi, thank you very much for replying to me. I would like to know how to calculate the average Dice and Iou about the dataset, not the corresponding loss.

**Comment by closertt** [comment]
Hi, I have another problem, when the training set is replaced with my own data, I get the error below.
![image](https://github.com/atlan-antillia/Image-Segmentation-Gastrointestinal-Polyp/assets/97007679/6d896a02-7725-4c54-a817-b21ac40184cf)
![image](https://github.com/atlan-antillia/Image-Segmentation-Gastrointestinal-Polyp/assets/97007679/5854a69d-bfa4-47f9-be73-f311b374c4b0)

**Comment by atlan-antillia** [comment]
Hi.
Please check the following point.
**1 Have you installed python packaged by the following command?**
 > pip install -r requirements.txt
**2 Are image_datapath and mask_datapth of [train] section in your train_eval_infer.config right?**
  image_datapath = "./GastrointestinalPolyp/train/images/"
  mask_datapath  = "./GastrointestinalPolyp/train/masks/"
  Are these datapath correct?

**Comment by closertt** [comment]
Hello, I solve a problem, I added create_backup = True in the .config file. But there is another problem, my path is correct, but I still get None values not supported error.

**Comment by atlan-antillia** [comment]
Hi.

my path is correct, but I still get None values not supported error.

—
If possible, could you please show me your error in full text (console
output)
to identity the reason of your "getting None values"?

2023年6月25日(日) 20:56 closertt ***@***.***>:
